src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -876,6 +876,12 @@ private void AddPatchNode(Patch patch, Vector2 position)
 
         node.Enabled = patchEnableStatusesToBeApplied?[patch] ?? true;
 
+        foreach (var effect in Enum.GetValues(typeof(WorldEffectVisuals)))
+        {
+            if (patch.ActiveDisplayVisuals.Contains((WorldEffectVisuals)effect))
```

Comment:
I think a more extensible code would be to just have `patch.ApplyActiveEffectVisuals(node)` and then put the more complex code into the patch class.